src/modules/bookings/dto/create-booking.dto.ts

```diff
@@ -0,0 +1,21 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsNotEmpty, IsUUID, IsDateString } from 'class-validator';
+import { Type } from 'class-transformer';
+
+export class CreateBookingDto {
+  @ApiProperty({ description: 'ID of the accommodation being booked' })
+  @IsUUID()
+  @IsNotEmpty()
+  @Type(() => String)
+  accommodationId: string;
+
+  @ApiProperty({ description: 'Start date of the booking in ISO format' })
+  @IsDateString()
+  @IsNotEmpty()
+  startDate: string;
+
+  @ApiProperty({ description: 'End date of the booking in ISO format' })
+  @IsDateString()
+  @IsNotEmpty()
+  endDate: string;
```

Comment:
Don't you think we need to check if startDate is earlier than endDate? So that we have protection against startDate coming later than endDate in case frontend fails to validate. What do you say?

Reply:
Yeah that makes sense